clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2894,6 +2912,22 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::update_next_record(output_ordinal
         // sched_type_t::MAP_AS_PREVIOUSLY).
         return;
     }
+    // We modify the tid and pid fields to ensure uniqueness for
+    // core-sharded-on-disk and with analyzers not using our workload identifiers.
+    // To maintain the original values, we write the workload ordinal into the top
+    // 32 bits.  We don't support distinguishing for 32-bit-build record_filter.
+    // We also ignore complexities on Mac with its 64-bit tid type.
+    int64_t workload = get_workload_ordinal(output);
+    memref_tid_t cur_tid;
+    if (record_type_has_tid(record, cur_tid) && workload > 0) {
```

Comment:
memtrace_stream_t::get_workload_id and memtrace_stream_t::get_input_id documentation should be updated to remove the "unlike ..." https://github.com/DynamoRIO/dynamorio/blob/ebe485dc60dc7106b2bf5782f02eb57354b2cb04/clients/drcachesim/common/memtrace_stream.h#L250

Reply:
Done.